actors/miner/tests/fip0081_initial_pledge.rs

```diff
@@ -0,0 +1,232 @@
+        &circulating_supply(),
+        50,
+        100,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1725).div_floor(10000),
+        initial_pledge
+    );
+}
+
+// On-ramp where 'baseline power' (85%).
+#[test]
+fn initial_pledge_after_ramp() {
+    let initial_pledge = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        150,
+        100,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1950).div_floor(10000),
+        initial_pledge
+    );
+}
+
+// Post-ramp where 'baseline power' has reduced effect (97%).
+#[test]
+fn initial_pledge_on_ramp_early() {
+    let initial_pledge = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        10,
+        100,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1545).div_floor(10000),
+        initial_pledge
+    );
+}
+
+// Post-ramp, first epoch, pledge should be 97% 'baseline' + 3% simple.
+#[test]
+fn initial_pledge_on_ramp_step() {
+    let initial_pledge = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        1,
+        10,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1545).div_floor(10000),
+        initial_pledge
+    );
+}
+
+// Validate pledges 1 epoch before and after ramp start.
+#[test]
+fn initial_pledge_ramp_edges() {
+    let initial_pledge_before_ramp = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        -1,
+        10,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1500).div_floor(10000),
+        initial_pledge_before_ramp
+    );
+
+    let initial_pledge_at_ramp = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        0,
+        10,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1500).div_floor(10000),
+        initial_pledge_at_ramp
+    );
+
+    let initial_pledge_on_ramp = initial_pledge_for_power(
+        &qa_sector_power(),
+        &StoragePower::from(1u64 << 37),
+        &reward_estimate(),
+        &power_estimate(),
+        &circulating_supply(),
+        1,
+        10,
+    );
+    assert_eq!(
+        TokenAmount::from_atto(1) + TokenAmount::from_whole(1545).div_floor(10000),
+        initial_pledge_on_ramp
+    );
+}
+
```

Comment:
nit: it would be nice to confirm that when we are less than 1/1000th of the way through the ramp that gamma == 1.  ramp duration == 2000, epochs since ramp == 1, pledge should be 1500

Reply:
I think we're also missing a case for "at ramp", e.g. `10, 10`, giving us `1950`, and similar to @ZenGround0's case, 999/1000, what does that give us? `1950` or one step before that?